File: front/src/services/projetoService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/projetos';

export default {
  getProjetos(page = 0, size = 10) {
    return axios.get(API_URL, {
      params: {
        page: page,
        size: size
      }
    });
  },

  createProjeto(projeto) {
    return axios.post(API_URL, projeto)
      .then(response => response.data)
      .catch(error => {
        throw error;
      });
  },

  updateProjeto(projeto) {
    return axios.put(`${API_URL}/${projeto.id}`, projeto)
      .then(response => response.data)
      .catch(error => {
        throw error;
      });
  },

  deleteProjeto(id) {
    return axios.delete(`${API_URL}/${id}`)
      .then(response => response.data)
      .catch(error => {
        throw error;
      });
  }
};

File: back/src/main/java/com/agcapital/service/ProjetoService.java
package com.agcapital.service;

import java.util.Optional;
import java.util.UUID;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import com.agcapital.domain.Projeto;
import com.agcapital.domain.Status;
import com.agcapital.exception.ProjetoNotFoundException;
import com.agcapital.exception.ProjetoWithActivityException;
import com.agcapital.repository.ProjetoRepository;

@Service
public class ProjetoService {

    @Autowired
    private ProjetoRepository projetoRepository;

    public Page<Projeto> getAllProjetos(int page, int size) {
    	Pageable pageable = PageRequest.of(page, size, Sort.by("nome").ascending());
	    Page<Projeto> projetos = projetoRepository.findAll(pageable);
	    return projetos;
    }

    public Optional<Projeto> getProjetoById(UUID id) {
        return projetoRepository.findById(id);
    }

    public Projeto createProjeto(Projeto projeto) {
        return projetoRepository.save(projeto);
    }

    public Projeto updateProjeto(UUID id, Projeto projeto) {
        Projeto projetoOld = projetoRepository.findById(id).orElseThrow(() -> new ProjetoNotFoundException());
        projetoOld = projeto;
    	return projetoRepository.save(projetoOld);
        
        
    }

    public void deleteProjeto(UUID id) {
    	Projeto projeto = projetoRepository.findById(id).orElseThrow(() -> new ProjetoNotFoundException());
    	boolean possuiTarefasEmAndamento = projeto.getAtividades().stream().anyMatch(atividade -> atividade.getStatus() != Status.FINALIZADO);
    	if(possuiTarefasEmAndamento)
    		throw new ProjetoWithActivityException();
    	
    	projetoRepository.delete(projeto);
    	
        
    }
}